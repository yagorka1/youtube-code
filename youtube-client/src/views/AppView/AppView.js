export default class AppView {
  constructor(titles) {
    this.titles = titles;
  }

  show() {
    const content = document.createElement('div');
    content.className = 'mainn';
    content.innerHTML = '<div class="searchField"><input type="text" class="teaxtarea"><input type="button" class="searchButton" value="search"></div><div class="cards"><div class="line"></div></div><div class="buttons"></div><div class="warningMessage"></div>';
    document.body.appendChild(content);
    window.console.log(this.titles);
    const message = document.getElementsByClassName('warningMessage')[0];
    message.innerHTML = '';
    const btn = document.getElementsByClassName('buttons')[0];
    btn.innerHTML = "<div class='buttons' ><i class='prev'><img class='arrow' src='../../assets/images/back.png'></i><i><p class='page'>1</p></i><i class='next'><img class='arrow next' src='../../assets/images/next.png'></i></div>";
    btn.classList.add('hid');
  }

  showClips() {
    const cards = document.getElementsByClassName('line')[0];
    cards.innerHTML = this.titles.map(title => `<a href="https://www.youtube.com/watch?v=${title[5]}" target="_blank">
    <div class="card">
    <div class="name"><h4>${title[0]}</h4></div><img src="${title[4]}"></img>
    <div class="data"><h5>${title[1]}</h5><br><span class="views">${title[6][0]} views, ${title[2].slice(0, 10)}</span><br>${title[3].slice(0, 100)}...<br>
    <br></div></div></a>`).join('');
    const message = document.getElementsByClassName('warningMessage')[0];
    message.classList.add('hid');
    const btn = document.getElementsByClassName('buttons')[0];
    btn.classList.remove('hid');
  }

  addClips() {
    const cards = document.getElementsByClassName('line')[0];
    const a = document.createElement('div');
    a.innerHTML = this.titles.map(title => `<a href="https://www.youtube.com/watch?v=${title[5]}" target="_blank">
    <div class="card">
    <div class="name"><h4>${title[0]}</h4></div><img src="${title[4]}"></img>
    <div class="data"><h5>${title[1]}</h5><br><span class="views">${title[6][0]} views, ${title[2].slice(0, 10)}</span><br>${title[3].slice(0, 100)}...<br>
    <br></div></div></a>`).join('');
    cards.appendChild(a);
    const message = document.getElementsByClassName('warningMessage')[0];
    message.classList.add('hid');
    const btn = document.getElementsByClassName('buttons')[0];
    btn.classList.remove('hid');
  }

  error() {
    window.console.log(this.titles);
    const cards = document.getElementsByClassName('line')[0];
    cards.innerHTML = '';
    const btn = document.getElementsByClassName('buttons')[0];
    btn.classList.add('hid');
    const message = document.getElementsByClassName('warningMessage')[0];
    message.classList.remove('hid');
    message.innerHTML = '<p class="txtMessage">По запросу ничего не найдено.</p><p class="txtMessage"> Рекомендации:</p> <ul class="items"><li>Убедитесь, что все слова написаны без ошибок.</li><li>Попробуйте использовать другие ключевые слова.</li><li> Попробуйте использовать более популярные ключевые слова.</li><li>Попробуйте уменьшить количество слов в запросе.</li>';
  }
}
