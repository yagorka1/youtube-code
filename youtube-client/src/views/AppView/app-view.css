body {
  margin: 0 auto;
}

.mainn {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  margin: 0 auto;
  width: 100%;
}

.searchField {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 70px;
  background-color: rgb(202, 201, 201);
  margin: 0 auto;
  box-shadow: 0 0 10px rgba(0,0,0,0.5);
}

.teaxtarea {
  width: 70%;
  height: 40%;
  background-color: rgb(238, 238, 238);
  font-size: 16px;
  padding-left: 10px;
  border: 1px solid gray;
}

.teaxtarea:hover {
  box-shadow: 0 0 10px rgba(0,0,0,0.5);
}

.teaxtarea:focus{
  outline: none;
}

.searchButton {
  height: 46%;
  margin-left: 10px;
}

.searchButton:hover {
  box-shadow: 0 0 10px rgba(0,0,0,0.5);
  cursor: pointer;
}

.cards {
  width: 91.2%;
  margin-top: 50px;
  margin-left: 3px;
  overflow: hidden;

}

.line {
  padding-top: 0px;
  position: relative;
  height: 500px;
  overflow: hidden;
  left: 0;
  transition: all ease 3s;
}

.card {
  background-color: rgb(210, 219, 75);
  width: 320px;
  height: 450px;
  float: left;
  border-radius: 5px;
  margin-bottom: 50px;
  margin-top: 15px;
  box-shadow: 0 0 10px rgba(0,0,0,0.5);
  padding-top:10px;
  padding-bottom:10px;
  margin-left: 25px;
  overflow: hidden;
}

.card:hover {
  background-color: red;
  transform: rotate(-5deg);
  color: white;
}

img {
  width: 320px;
  height: 180px;
  margin: 0 auto;
}

a {
  text-decoration: none;
  color: black;
}

.name {
  height: 70px;
  margin: 10px;
}

h4 {
  margin: 0 auto;
}

.data {
  width: 90%;
  height: 180px;
  margin: 10px;
  line-height: 16pt;
}

h5 {
  margin: 0 auto;
  margin-bottom: -15px;
  font-size: 12pt;
  font-weight: 500;
}

.views {
  line-height: 34pt;
}

.buttons {
  margin-top: 15px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  width: 200px;
}

.arrow {
  width: 30px;
  height: 30px;
  display: inline-block;
}

.arrow:hover {
 background-color: red;
 cursor: pointer;
}

.warningMessage{
  display: flex;
  flex-direction: column;
  font-size: 18px;
}

.txtMessage {
  margin-top: -250px;
  margin-bottom: 10px;
  z-index: 1;
}

.items {
  list-style: disc;
  margin-left: -10px;
}

.hid {
  display: none;
}

.page {
  margin-top: 2px;
  width: 5px;
  font-style: normal;
  font-size: 24px;
}