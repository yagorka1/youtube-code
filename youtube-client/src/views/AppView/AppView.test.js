import AppView from './AppView';

describe('AppView.prototype.show', () => {
  it('Should be an instance of Function', () => {
    expect(AppView.prototype.show).toBeInstanceOf(Function);
  });

  it('Should be show correctly', () => {
    const context = {
      titles: [
        'Video about JS',
        'Another video ...',
        'I need more video',
      ],
    };

    AppView.prototype.show.call(context);
    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
