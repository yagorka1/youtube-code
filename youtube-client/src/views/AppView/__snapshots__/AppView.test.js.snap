// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppView.prototype.show Should be show correctly 1`] = `"<ul><li>Video about JS</li><li>Another video ...</li><li>I need more video</li></ul>"`;
